### PR TITLE
Test a rebooted system with the enabled initial setup 

### PR DESCRIPTION
**Pull request by poncovka**
It doesn't have to be pretty, it just needs to work.

Process the `additional_runner_args` bash function again to allow to specify
additional arguments for the `virt-install` tool. Remove the deprecated usage
of this function from the `ibft` test.

Test the reboot of the installed system with and without the initial setup.
Verify that the installed system boots without any errors or tracebacks.

Next steps:
* Optimize the initial setup tests for RHEL.
* Install the same version of anaconda as on the boot.iso.
* Apply the same updates image as on the boot.iso.

### CODE REVIEW

File: fragments/shared/validation/success_on_first_boot.ks
Comment (jstodola) [inline fragments/shared/validation/success_on_first_boot.ks:59]: Wouldn't be simpler to run the %post script in chroot?

File: reboot-initial-setup-gui.ks.in
Comment (jstodola) [inline reboot-initial-setup-gui.ks.in:32]: The test will pass if initial-setup is not started at all for whatever reason. The same for tui.

File: reboot.ks.in
Comment (jstodola) [inline reboot.ks.in:20]: The test will fail due to any error in journal (for example from kernel) - is it really wanted?

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks:29]: Fixed.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks]: Fixed.

File: reboot-initial-setup-gui.ks.in
Comment (poncovka) [reply reboot-initial-setup-gui.ks.in:36]: Fixed.

File: reboot.ks.in
Comment (poncovka) [reply reboot.ks.in]: Fixed.

**Comment by poncovka** [comment]
Updated.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (jstodola) [reply fragments/shared/validation/success_on_first_boot.ks:29]: # IMPORTANT: Variables needs to be escaped!
They don't need to be escaped any more!

**Comment by jstodola** [comment]
It looks good to me (but I have not tried to execute the tests).

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks]: Fixed. Thanks!

**Review by rvykydal — DISMISSED** [review]
Looks great to me.

**Review by M4rtinK — APPROVED** [review]
Looks good to me, thanks a bunch! :)

Like really, this is perfect & has a big potential for not just smoke testing Initial Setup but other complex cases where we can't just statically verify data on disk to estimate if the system will boot correctly or not. At least boot loader installations and verification of complex network configs should be able to use this out of the box IMHO. :)

So indeed, very much appreciated! :)

File: fragments/shared/validation/success_on_first_boot.ks
Comment (jkonecny12) [inline fragments/shared/validation/success_on_first_boot.ks:49]: I don't mind this method but it would be great to have a separator to find out where the old ends and what is the new part.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (jkonecny12) [inline fragments/shared/validation/success_on_first_boot.ks:29]: I wonder why you have `'` around the `EOF` here.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (jkonecny12) [inline fragments/shared/validation/success_on_first_boot.ks:1]: Please add another header comment that the file `/usr/libexec/kickstart-test.sh` should be created by the test.

**Review by jkonecny12 — DISMISSED** [review]
Looks good to me. I just find a few nitpicks which are not blocking.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks:49]: `journalctl` prints by default `-- Logs begin ...` on the first line.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks:29]: Please, see the comment https://github.com/rhinstaller/kickstart-tests/pull/464#discussion_r558914963.

File: fragments/shared/validation/success_on_first_boot.ks
Comment (poncovka) [reply fragments/shared/validation/success_on_first_boot.ks:1]: Fixed.

**Comment by poncovka** [comment]
Btw. I have unchecked "Dismiss stale pull request approvals when new commits are pushed".

**Comment by poncovka** [comment]
The initial setup tests are timing out because of the bug that will be fixed in https://github.com/rhinstaller/anaconda/pull/3105.
```
TEST                                                    | RESULT     | EXPLANATION
--------------------------------------------------------+------------+----------------------
container                                               | SUCCESS    | 
bindtomac-network-device-default-httpks                 | SUCCESS    | 
reboot                                                  | SUCCESS    | 
reboot-initial-setup-gui                                | TIMED OUT  | 
reboot-initial-setup-tui                                | TIMED OUT  | 
```

**Review by jkonecny12 — APPROVED** [review]
LGTM!

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me, thank you!

**Comment by poncovka** [comment]
The `reboot-initial-setup-tui` test will be fixed with the next compose. Merging.

**Pull request merged**
